src/testdrive/src/action.rs

```diff
@@ -120,6 +125,93 @@ impl State {
+            // you can't delete a bucket that has stuff in it
+            let mut continuation_token = None;
+            loop {
+                let response = self
+                    .s3_client
+                    .list_objects_v2(ListObjectsV2Request {
+                        bucket: bucket.clone(),
+                        continuation_token: continuation_token.take(),
+                        ..Default::default()
+                    })
+                    .await;
+                match response {
+                    Ok(response) => {
+                        if let Some(objects) = response.contents {
+                            for obj in objects {
+                                if let Err(e) = self
+                                    .s3_client
+                                    .delete_object(DeleteObjectRequest {
+                                        bucket: bucket.clone(),
+                                        key: obj.key.clone().unwrap(),
+                                        ..Default::default()
+                                    })
+                                    .await
+                                {
+                                    errors.push(
+                                        Error::with_cause(
+                                            format!(
+                                                "Deleting object {}/{}",
+                                                bucket,
+                                                obj.key.unwrap()
+                                            ),
+                                            e,
+                                        )
+                                        .into(),
+                                    );
+                                }
+                            }
+                        }
+
+                        if response.next_continuation_token.is_none() {
+                            break;
+                        }
+                        continuation_token = response.next_continuation_token;
+                    }
+                    Err(e) => {
+                        errors.push(
+                            Error::with_cause(format!("Listing objects for bucket {}", bucket), e)
+                                .into(),
+                        );
+                    }
+                }
+            }
+
+            let res = self
+                .s3_client
+                .delete_bucket(DeleteBucketRequest {
+                    bucket,
+                    expected_bucket_owner: None,
+                })
+                .await;
+
+            if let Err(e) = res {
+                errors.push(e.into());
+            }
+        }
+
+        if errors.is_empty() {
+            Ok(())
+        } else {
+            Err(Error::General {
+                ctx: format!("deleting S3 buckets: {} errors", errors.len()),
+                causes: errors.into_iter().map(Into::into).collect(),
```

Comment:
You already called `into` when you pushed into the vec, so you can just do `causes: errors` here.